polaris/hub/client.py

```diff
@@ -832,4 +831,30 @@ def list_competitions(
         )
         benchmarks_list = [f"{HubOwner(**bm['owner'])}/{bm['name']}" for bm in response["data"]]
         return benchmarks_list
-    
\ No newline at end of file
+
+    def evaluate_competition(
+        self,
+        competition: CompetitionSpecification,
+        y_pred: PredictionsType
+    ) -> BenchmarkResults:
+        """Evaluate the predictions for a competition on the Polaris Hub.
+
+        Args:
+            competition: The competition to evaluate the predictions for.
+            y_pred: The predictions for the test set, as NumPy arrays.
+                If there are multiple targets, the predictions should be wrapped in a dictionary with the target labels as keys.
+
+        Returns:
+             A `BenchmarkResults` object.
+        """
+        return self._base_request_to_hub(
+            url=f"/v2/competition/evaluate",
+            method="PUT",
+            json={
+                "competition": competition.model_dump(exclude_none=True,
+                                                      exclude=["dataset",
+                                                               "split"],
+                                                      by_alias=True),
```

Comment:
Just revisiting our thoughts on what we need to send over the wire here. This approach works, but I'm more in favour of just sending the competition identifier (owner/name) along with the predictions.

Using the identifier, we can then call the Hub server side to get the necessary data we need into the evaluation service. 

Once we decide on a server to server auth mechanism, we'll be able to implement that!

Reply:
This makes so much more sense than trying to reconstruct a valid `CompetitionSpecification` on the hub side -- I was initially trying to avoid having to figure out how to auth and access the db from the python endpoints run as verbal serverless functions, but you're right that we'll have to figure that out anyway regardless, so may as well do it here. It'll simplify a lot of things!